lib/restforce/concerns/sobject_collection_api.rb

```diff
@@ -0,0 +1,149 @@
+# frozen_string_literal: true
+
+require 'restforce/concerns/verbs'
+
+module Restforce
+  module Concerns
+    module SObjectCollectionAPI
+      extend Restforce::Concerns::Verbs
+
+      define_verbs :post, :patch, :delete
+
+      def collection_get(sobject_name, ids, fields)
+        raise ArgumentError, "ids are required" if Array(ids).empty?
+        raise ArgumentError, "fields are required" if Array(fields).empty?
+
+        api_get("composite/sobjects/#{sobject_name}",
+                ids: ids.join(','),
+                fields: fields.join(',')).body
+      end
+
+      def collection_delete(ids, opts = {})
+        all_or_none = opts.delete(:all_or_none) || false
+        raise ArgumentError, "ids are required" if Array(ids).empty?
+
+        results = api_delete("composite/sobjects",
+                             ids: ids.join(','),
+                             allOrNone: all_or_none).body
+        CollectionResponse.new(results, all_or_none: all_or_none).response
+        results
+      end
+
+      def collection_delete!(*ids)
+        collection_delete(ids.flatten, all_or_none: true)
+      end
+
+      def collection_create(opts = {})
+        all_or_none = opts.delete(:all_or_none) || false
+        builder = RecordsBuilder.new
+        yield(builder)
+        return builder.records if opts[:debug]
```

Comment:
What do you think of using `dry_run` here too?

```suggestion
        return builder.records if opts[:dry_run]
```